### DISCUSSION TITLE
Why can't the key of the returned value start with “_sa”?

### DISCUSSION THREAD

**Discussion by leafotto** [fastapi/fastapi#12059] [Questions] [answered]
### Privileged issue

- [X] I'm @tiangolo or he asked me directly to create an issue here.

### Issue Content

```
import uvicorn
from fastapi import FastAPI


app = FastAPI()


@app.get("/")
def root():
    return {"_sa": "Hello World", "status": "OK"}


if __name__ == '__main__':
    uvicorn.run(app, host="0.0.0.0", port=8000)
```
The result of the above code is:
```
{
    "status": "OK"
}
```

**Comment by PhysicallyActive** [comment]
If I'm not mistaken this is not a FastAPI issue, but rather a Pydantic feature.
According to [this stackoverflow thread](https://stackoverflow.com/a/75712642) Pydantic V2 considers parameters prefixed with an underscore as "PrivateAttr", i.e., private attributes.

**Comment by leafotto** [comment]
> 如果我没记错的话，这不是 FastAPI 问题，而是 Pydantic 功能。根据[这个 stackoverflow 线程](https://stackoverflow.com/a/75712642)，Pydantic V2 将带有下划线前缀的参数视为“PrivateAttr”，即私有属性。

可以修改以上代码做个实验：
```
def root():
    result =  {"_private": "test", "_sa": "Hello World", "status": "OK"}
    print(result)
    return result
```
我测试的结果是：
```
{
    "_private": "test",
    "status": "OK"
}
```

**Reply by sinisaos** [reply, marked as answer]
@leafotto The reason for this is that the `jsonable_encoder` (which FastAPI uses internally to convert data) does not allow any field that [starts with the name `_sa`](https://github.com/fastapi/fastapi/blob/d0ce9d2bdf7f3c78b6f193288b9925b952da27c9/fastapi/encoders.py#L276) which is explained [here](https://github.com/fastapi/fastapi/blob/master/fastapi/encoders.py#L183-L187)
